src/components/FormElements/XdrLedgerKeyPicker.tsx

```diff
@@ -276,17 +278,14 @@ export const XdrLedgerKeyPicker = ({
     [reset, xdrDecodeLedgerKeyToJson],
   );
 
-  // Render input fields on mount if there is Ledger Key XDR value
+  // Render input fields based on xdr value
+  // This also gets updated when an item is deleted
   useEffect(() => {
-    if (value && !ledgerKeyXdrToJsonString && !selectedLedgerKey) {
+    if (value) {
       validateLedgerKeyXdr(value);
     }
-  }, [
-    value,
-    ledgerKeyXdrToJsonString,
-    validateLedgerKeyXdr,
-    selectedLedgerKey,
-  ]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This is not needed anymore.

Reply:
it asked to add `validateLedgerKeyXdr` and I didn't want to add a function in there. wdyt

Reply:
Ahh, I see. Could we add a comment like [here](https://github.com/stellar/laboratory/blob/main/src/app/(sidebar)/account/fund/page.tsx#L73) saying what is excluded? It seems more intentional that way.